logical/src/main/java/org/apache/drill/common/expression/SchemaPath.java

```diff
@@ -115,6 +112,33 @@ public static SchemaPath create(NamePart namePart) {
   }
 
   /**
+   * Parses input string and returns {@code SchemaPath} instance.
+   *
+   * @param expr input string to be parsed
+   * @return {@code SchemaPath} instance
+   */
+  public static SchemaPath parseFromString(String expr) {
```

Comment:
What are the rules for parsing a string containing dots? Containing []?

Reply:
Thanks for the explanation. Can you copy it into the Javadoc for this method? Will help others in the future.